app/views/posts/_post_presenter_index.html.erb

```diff
@@ -0,0 +1,9 @@
+<div>
+  <% if post.url_post? %>
+    <span><%= link_to(post.title, post.url) %></span>
+  <% else %>
+    <span><%= post.text.truncate(80, {separator: " "}) %></span>
+  <% end %>
```

Comment:
what if instead of a conditional, `post` had a `type` method that returned either `url` or `text` and that type was used to render a specific partial?  something like:

```erb
# _post_presenter_index.html.erb
<%= render partial: "#{post.type}_post_content", locals: { post: post } %>

# _url_post_content.html.erb
<span><%= link_to(post.title, post.url) %></span>

# _text_post_content.html.erb
<span><%= post.text.truncate(80, {separator: " "}) %></span>
```

A bit more flexible if more types were added in the future.

Reply:
I like that idea